pkg/kv/kvserver/store_pool.go

```diff
@@ -174,15 +174,18 @@ func LivenessStatus(
 		}
 		return livenesspb.NodeLivenessStatus_DEAD
 	}
-	if !l.Membership.Active() {
-		return livenesspb.NodeLivenessStatus_DECOMMISSIONING
-	}
-	if l.Draining {
-		return livenesspb.NodeLivenessStatus_DRAINING
-	}
 	if l.IsLive(now) {
+		if !l.Membership.Active() {
+			return livenesspb.NodeLivenessStatus_DECOMMISSIONING
+		}
+		if l.Draining {
+			return livenesspb.NodeLivenessStatus_DRAINING
+		}
 		return livenesspb.NodeLivenessStatus_LIVE
 	}
+	if !l.Membership.Active() {
+		return livenesspb.NodeLivenessStatus_DECOMMISSIONED
```

Comment:
This seems incorrect, I think this should be UNAVAILABLE as well.

Reply:
I went back and forth on that one, the reason I thought it should be decommissioned was that a node would go Decommissioning -> Decommissioned (Unavailable) -> Decommissioned (Dead) this way, if its shutdown in the middle of the process. No strong feelings on this, so will change it.

Reply:
The reason I think this isn't quite right is that while a node is decommissioning, it typically still has data and so when it goes down that is a problem. A `decommissioned` node is one where you don't care any more whether it's down.

Thanks for making the change.